src/classes/TB_StartEndTime_TDTM.cls

```diff
@@ -0,0 +1,164 @@
+
+            // This shouldn't ever happen, but check it just to be safe.
+            if (oldTimeBlock == null) {
+                continue;
+            }
+
+            if (timeBlock.Start_Time__c != oldTimeBlock.Start_Time__c || timeBlock.End_Time__c != oldTimeBlock.End_Time__c) {
+                timeBlocksWithUpdatedTimes.add(timeBlock.Id);
+            }
+        }
+
+        return timeBlocksWithUpdatedTimes;
+    }
+
+    /*******************************************************************************************************
+    * @description Queries for Course Offering Schedules related to the Time Blocks that have an updated
+    * Start Time and/or End Time and updates the Start Time and End Time on the Course Offering Schedules to
+    * match the new values.
+    * @param timeBlocksWithUpdatedTimes The Set of Time Block IDs that have an updated Start Time and/or End Time.
+    * @param timeBlocksMap The map of updated Time Blocks.
+    * @param oldTimeBlocksMap The map of old Time Block records prior to the update.
+    * @return The updated Course Offering Schedules.
+    ********************************************************************************************************/
+    private List<Course_Offering_Schedule__c> getCourseOfferingSchedulesToUpdate(Set<Id> timeBlocksWithUpdatedTimes, Map<Id, Time_Block__c> timeBlocksMap, Map<Id, Time_Block__c> oldTimeBlocksMap) {
+        List<Course_Offering_Schedule__c> courseOfferingSchedulesToUpdate = new List<Course_Offering_Schedule__c>();
+
+        // Query for all related Course Offering Schedules.
+        List<Course_Offering_Schedule__c> courseOfferingSchedules = [SELECT Start_Time__c, 
+                                                                            End_Time__c, 
+                                                                            Time_Block__c 
+                                                                       FROM Course_Offering_Schedule__c 
+                                                                      WHERE Time_Block__c IN :timeBlocksWithUpdatedTimes];
+        
+        for (Course_Offering_Schedule__c courseOfferingSchedule : courseOfferingSchedules) {
+            Time_Block__c timeBlock = timeBlocksMap.get(courseOfferingSchedule.Time_Block__c);
+            Time_Block__c oldTimeBlock = oldTimeBlocksMap.get(courseOfferingSchedule.Time_Block__c);
+
+            // This shouldn't ever happen, but check it just to be safe.
+            if (timeBlock == null || oldTimeBlock == null) {
+                continue;
+            }
+
+            // Does the Course Offering Schedule Start Time and End Time align with the previous Time Block Start Time and End Time?
+            // It should because we have TDTM logic that prevents this from happening. If so, update the Course Offering Schedule to
+            // match the new values.
+            Boolean updated = false;
```

Comment:
only question I have here is "is this overkill" -- can we get away with just setting the start/end times based on the time block and move on? I realize we may be updating something that doesn't need to be updated (but if we are changing the timeblock start/end times then they'll need to be updated anyway right?), but might reduce code footprint down to: 

`courseOfferingSchedule.Start_Time__c = timeblock.Start_Time__c;`
`courseOfferingSchedule.End_Time__c = timeblock.End_Time__c;`

Again, not a hill I'm willing to die on here either.. :)

Reply:
I didn't necessarily want this to be complicated, but if for some reason the customer had disabled the COS_StartEndTime_TDTM class in order to import data and the Course Offering Schedule times didn't align with the Time Block times, I wouldn't want that data to get overwritten if they changed the Time Block to something else. I'd rather them get an error when they try to update the Course Offering Schedule that tells them to correct their data.